fem/tmop_tools.cpp

```diff
@@ -29,13 +29,28 @@ void AdvectorCG::SetInitialField(const Vector &init_nodes,
 void AdvectorCG::ComputeAtNewPosition(const Vector &new_nodes,
                                       Vector &new_field)
 {
-#if defined(MFEM_DEBUG) || defined(MFEM_USE_MPI)
-   int myid = 0;
-#endif
-   Mesh *m = mesh;
+   // This function will not work for AMR meshes in the current state.
+   const int dim     = fes->GetFE(0)->GetDim(),
+             ncomp   = fes->GetVDim(),
+             pnt_cnt = new_nodes.Size()/dim;
+
+   new_field = field0;
+
+   for (int i = 0; i < ncomp; i++)
+   {
+      Vector new_field_temp(new_field.GetData()+i*pnt_cnt, pnt_cnt);
+      ComputeAtNewPositionScalar(new_nodes, new_field_temp);
+   }
 
+   //field0 = new_field;
+   //nodes0 = new_nodes;
```

Comment:
Why are these commented out? That would change the previous result significantly.